ray-operator/controllers/ray/common/association_test.go

```diff
@@ -149,6 +149,113 @@ func TestRayClusterHeadlessServiceListOptions(t *testing.T) {
 	}
 }
 
+func TestRayClusterHeadPodsListOptions(t *testing.T) {
```

Comment:
The tests can't ensure that the functions work as expected. It is very easy to forget to update `listOptions` below if we update labels. How about create a test in `raycluster_controller_test.go` to test whether the functions can get the correct number of Pods? The test will create a RayCluster with 2 worker groups so that we can test the four association functions above.

Reply:
Instead of creating a new test, how about replacing the usages of MatchingLabels with the new AssociationOption in existing tests?

Reply:
We should use the new `AssociationOption` to associate anything in the codebase (not necessary to include all of them in this PR). 

>  I found the existing tests can cover all the four association functions above.

Which existing tests are you referring to? Do you mean envtest or others? I think envtest is necessary.

Reply:
Yes, the existing envtests in the raycluster_controller_test.go.
I refactored them in the latest commit https://github.com/ray-project/kuberay/pull/2023/commits/8e7a7c3c0a2aac4336e5dc0a90cf5947757b3962

Reply:
Nice!